### ISSUE TITLE
Cleanup and streamline our integration test bases

### ISSUE THREAD

**Issue by jathanism** [nautobot/nautobot#815]
### Proposed Changes

I propose the following for integration tests cases:

- Tests using `SeleniumTestCase` as their base be replaced to `SplinterTestCase`

```
$ egrep -r "SeleniumTestCase\)" nautobot | grep -v md
nautobot/core/tests/integration/test_home.py:class HomeTestCase(SeleniumTestCase):
nautobot/core/tests/integration/test_plugin_home.py:class PluginHomeTestCase(SeleniumTestCase):
nautobot/core/tests/integration/test_navbar.py:class NavBarTestCase(SeleniumTestCase):
nautobot/core/tests/integration/test_plugin_navbar.py:class PluginNavBarTestCase(SeleniumTestCase):
nautobot/extras/tests/integration/test_plugin_banner.py:class PluginBannerTestCase(SeleniumTestCase):
```

- Tests using `CeleryTestCase` as their base be replaced with `SplinterTastCase` and also setting `requires_celery = True` as a class variable (Will be merged in https://github.com/nautobot/nautobot/pull/790).

```
$ egrep -r "CeleryTestCase\)" nautobot
nautobot/extras/tests/test_customfields.py:class CustomFieldBackgroundTasks(CeleryTestCase):
```

- Migrate any internal calls to `self.selenium` to use `self.browser` (the Splinter client)
- Once migration work is complete, remove `CeleryTestCase`, `SeleniumTestCase`, and `NautobotRemote` (and anything else no longer required)
- Finally, rename `SplinterTestCase` to `SeleniumTestCase` 

### Justification

Our integration test suite isn't that old but is already experiencing a little bit of tech debt. This is due to two key factors:

##### 1. Selenium vs. Splinter

We had already introduced `SeleniumTestCase` using the default Selenium Python library. Selenium is a very thin veil over the underlying Selenium Java API, so it's not very Pythonic and is often overly verbose so we also started implementing custom helper methods to wrap these calls. 
 
We decided to instead start using the [`splinter`](https://splinter.readthedocs.io/en/latest/) library which is more Pythonic. and introduced the `SplinterTestCase` that uses it. 

##### 2. We introduced Celery

Celery requires special cases for testing because of the way its workers do threading. It is required to be a `TransactionTestCase`, and fortunately the `LiveServerTestCase` that is the base of Selenium/Splinter classes also inherits from `TransactionTestCase`. So in #790 we made the decision since we were adding new tests to go ahead and merge `CeleryTestCase` w/ `SplinterTestCase` and gate whether or not the worker gets started using the `requires_celery = True` class variable.

**Comment by briddo** [comment]
As discussed with @jathanism and @glennmatthews, I've left out migrating tests which inherit from nautobot.utilities.testing.CeleryTestCase as these test cases don't actually use Selenium.

**Comment by bryanculver** [comment]
Merged into `next`.
